src/main/java/com/sourcegraph/cody/autocomplete/CodyEditorFactoryListener.java

```diff
@@ -186,7 +186,8 @@ private static Range getSelection(Editor editor) {
   }
 
   // Sends a textDocument/didChange notification to the agent server.
-  public static void informAgentAboutEditorChange(@Nullable Editor editor) {
+  public static void informAgentAboutEditorChange(
+      @Nullable Editor editor, boolean skipCodebaseOnFileOpened) {
```

Comment:
nit: I generally try to avoid boolean parameters like this where the majority of the call-sites pass the same value (`true` in this case here). One problem with this approach is that the method becomes difficult to reason about when you add a second boolean parameter the next time you have to customize the behavior even more. Another problem is that passing in a boolean `true` at call-site makes it difficult to understand what the `true` value does.

- A cheap workaround is to at add an overload without the boolean parameter.
- A better workaround is to rewrite this code into Kotlin so that you can use default parameter values and named argument.

We should aim to rewrite everything into Kotlin so I vote for the second option.

Reply:
moved to Kotlin! 🎉